lib/Dialect/TritonGPU/Transforms/Pipeliner/PipelineExpander.cpp

```diff
@@ -488,9 +493,14 @@ LogicalResult LoopPipelinerInternal::createKernel(
         continue;
       auto remap = loopArgMap.find(
           std::make_pair(operand->get(), useStage - stageDef->second));
-      assert(remap != loopArgMap.end());
-      nestedNewOp->setOperand(operand->getOperandNumber(),
-                              newForOp.getRegionIterArgs()[remap->second]);
+      if (remap != loopArgMap.end())
```

Comment:
I don't understand in what case this condition would be true. If the definition is outside the loop we should continue in the condition above as it wouldn't have a stage.

Reply:
Ah, let me fix lit tests first

Reply:
>Maybe we could upstream it to llvm as well if you think the fix is reasonable

yes definitely.